src/contribute/docker-images.md

```diff
@@ -0,0 +1,151 @@
+      -e SRC_FILE="v2.1.0-RC1.zip" \
+      -e SRC_DIR="beam-2.1.0-RC1" \
+      -it beam:openjdk8 /bin/bash
+
+*Notice that SRC_FILE is different from SRC_DIR because github replaces the 'v'
+character in the internal name of the zip*.
+
+To run a container with the source code during a vote:
+
+    docker run \
+      -e URL="https://dist.apache.org/repos/dist/dev/beam/2.1.0/apache-beam-2.1.0-source-release.zip" \
+      -e SRC_FILE="apache-beam-2.1.0-source-release.zip" \
+      -e SRC_DIR="apache-beam-2.1.0" \
+      -it beam:openjdk8 /bin/bash
+
+### Testing in an specific environment with your own source
+
+You can also overwrite the volume containing the source code of Beam from a
+directory in your host machine. This is useful to test your code in different
+versions of Java.
+
+If you have the code in the ~/workspace/beam directory and you want to quickly
+test it with Beam on Java 7 you can do:
+
+    docker run -v ~/workspace/beam:/home/user/beam -it beam:openjdk7 /bin/bash
+
+### Performance improvements
+
+The docker image does **not** contain the java dependencies and this could make
+the execution of the commands in the image take more time than expected. One way
+to speed this is by mounting a directory with the dependencies as a docker
+volume. The following command shows how to do this locally from the host machine
+mapping the local maven's `~/.m2` directory in the container.
+
+    docker run -v ~/.m2:/home/user/.m2 -it beam:openjdk8 /bin/bash
+
+You can also use an empty directory instead of `~/.m2` if you want to create a
+different environment from the local one to share it between runs.
+
+**Maven Improvements:**
+
+You can also speed the execution by adding environment variables to maven, for
+example the MAVEN_OPTS environment variable may speed up compilation:
+ 
+    docker run \
+      -e MAVEN_OPTS='-client -XX:+TieredCompilation -XX:TieredStopAtLevel=1 -Xverify:none' \
+      -it beam:openjdk8 /bin/bash
+
+Maven does not execute on multiple cores by default, to do this you have to
+enable the threading execution:
+    
+    mvn --threads 1C ...
+
+## Git based image
+
+It creates a docker container with a cloned version of the git repository, its
+branches and all the pull-requests. When it is run, it updates and checks out
+the specified branch/tag/pull-request.
+
+You can choose the branch to execute via the environment variable:
+
+`BRANCH`: The name of the specific branch, tag or pr to check out.
+
+By default it checks out master from the github repo, but if you want to check
+for example a specific pull request ID you can build the image like this (you
+must change ID for the number):
+
+    cd git/openjdk8
+    docker build -t beam:git .
+    docker run -e BRANCH=pr/ID -it beam:git /bin/bash
+
+## Release image
+
+The release image is a container with the needed requirements to run Beam out of
+the box in a container, or to submit a pipeline to execute on Google Cloud
+Dataflow to build it you just do:
```

Comment:
wondering about ending the sentence after Dataflow. "... Cloud Dataflow. To build it, run:"